app/src/main/java/Activities/ViewProblemActivity.java

```diff
@@ -25,29 +27,88 @@
 
 import java.util.ArrayList;
 
+import Controllers.BrowseUserProblemsController;
+import Controllers.ElasticsearchProblemController;
 import Controllers.OfflineLoadController;
 
-public class ViewProblemActivity extends AppCompatActivity {
+public class ViewProblemActivity extends AppCompatActivity{
+
+    protected TextView view_problem_title_text,
+        view_problem_date_text,
+        view_problem_description_text,
+        view_problem_numRecords_text;
+
+    protected Button setReminderButton,
+        viewRecordsButton,
+        viewMapButton,
+        viewSlideshowButton;
 
-    private TextView view_problem_title_text;
     private int position;
     private ArrayList<Problem> problems;
+    private Problem currentProblem;
+    private String userId;
+    private BrowseUserProblemsController problemController = new BrowseUserProblemsController();
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+
+    }
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_view_problem);
 
-        this.view_problem_title_text = (TextView)findViewById(R.id.view_problem_title_id);
-
+        //Extract selected problem object through intent and index of problem list
         Intent intent = getIntent();
         this.position = intent.getIntExtra("position",0);
```

Comment:
Instead of position, is it possible to pass problem title/id? and then find the current problem that way. Since I'm afraid that the wrong problem will be fetched if the database is changed during the intent passing.

Reply:
I just figured out you could use _adapter.getItem(position)_ to actually get the problem itself. Should've done this earlier.

Reply:
Thanks! I will edit it 👍

Reply:
Actually..I think adapter is just used for ListViews isn't it? There is no listview for this activity

Reply:
Oh, I meant to find the problem in browse problem activity,
problem = adapter.getItem(position)
and then instead putExtra("position",...) we could putExtra("problem.getTitle())
The problem's title/ id would be pass to view problem